Assets/MRTK/Core/Inspectors/Setup/MixedRealityProjectConfiguratorWindow.cs

```diff
@@ -2,66 +2,89 @@
 // Licensed under the MIT License.
 
 using Microsoft.MixedReality.Toolkit.Editor;
+using System;
 using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using TMPro;
 using UnityEditor;
 using UnityEditor.Compilation;
 using UnityEngine;
 using MRConfig = Microsoft.MixedReality.Toolkit.Utilities.Editor.MixedRealityProjectConfigurator.Configurations;
 
 namespace Microsoft.MixedReality.Toolkit.Utilities.Editor
 {
+
     public class MixedRealityProjectConfiguratorWindow : EditorWindow
     {
-        private readonly Dictionary<MRConfig, bool> trackToggles = new Dictionary<MRConfig, bool>()
+        private const float Default_Window_Height = 500.0f;
+        private const float Default_Window_Width = 300.0f;
+        private const string XRPipelineDocsUrl = "TO DO";
+        private const string XRSDKUnityDocsUrl = "https://docs.unity3d.com/Manual/configuring-project-for-xr.html";
+        private const string MSOpenXRPluginUrl = "https://aka.ms/openxr-unity-install";
+        private readonly GUIContent ApplyButtonContent = new GUIContent("Apply", "Apply configurations to this Unity Project");
+        private readonly GUIContent SkipButtonContent = new GUIContent("Skip", "Skip to next step");
+        private readonly GUIContent LaterButtonContent = new GUIContent("Later", "Do not show this pop-up notification until next session");
+        private readonly GUIContent IgnoreButtonContent = new GUIContent("Ignore", "Modify this preference under Edit > Project Settings > Mixed Reality Toolkit");
+
+        private static ConfigurationStage currentStage
         {
-            { MRConfig.ForceTextSerialization, true },
-            { MRConfig.VisibleMetaFiles, true },
-            { MRConfig.VirtualRealitySupported, true },
-            { MRConfig.OptimalRenderingPath, true },
-            { MRConfig.SpatialAwarenessLayer, true },
-            { MRConfig.AudioSpatializer, true },
+            get => MixedRealityProjectPreferences.ConfiguratorState;
+            set => MixedRealityProjectPreferences.ConfiguratorState = value;
+        }
 
-            // UWP Capabilities
-            { MRConfig.MicrophoneCapability, true },
-            { MRConfig.InternetClientCapability, true },
-            { MRConfig.SpatialPerceptionCapability, true },
+        public static MixedRealityProjectConfiguratorWindow Instance { get; private set; }
+
+        public static bool IsOpen => Instance != null;
```

Comment:
Can Instance ever be set to null? Double checking where how this variable is used even though this was the way we did it before.

Reply:
Ping on this @MaxWang-MS

Reply:
Instance cannot be set to null but by default it is null before initialization. We have static functions to show the window and we use this to see if there has been an instance of the window that we can reuse.